tests/async/test_async_queryset.py

```diff
@@ -62,6 +64,11 @@ async def test_aiterator_invalid_chunk_size(self):
                 async for m in qs:
                     pass
 
+    async def test_aiterator_prefetch_related(self):
+        async for s in SimpleModel.objects.prefetch_related("relatedmodel_set"):
+            count = await s.relatedmodel_set.acount()
+            self.assertEqual(count, 1)
```

Comment:
I thought about this point quite a bit while creating this test.  First, I wasn't able to use an "assertNumQueries", because the way it processes the connection object isn't valid in an async method.  Secondly, if the test does not execute a prefetch, then accessing the counts of the related manager would cause a lazy read, which would raise the "cannot execute a synchronous operation from an async context" error.  

I wasn't able to think of a test that was more precise, even after looking through other tests in the 'async' area.

Would you have any suggestions for improving this?

Reply:
> First, I wasn't able to use an "assertNumQueries", because the way it processes the connection object isn't valid in an async method.

Well that sucks :/

https://github.com/django/django/blob/9af0e9b3bade140fa3590b2f5831af932c35dc13/tests/async/test_async_queryset.py#L238-L240

I think that adding a similar TODO and making sure that `s` has the `_prefetched_objects_cache` set on it.